x/mongo/driver/batch_cursor.go

```diff
@@ -133,13 +140,15 @@ func NewCursorResponse(info ResponseInfo) (CursorResponse, error) {
 
 // CursorOptions are extra options that are required to construct a BatchCursor.
 type CursorOptions struct {
-	BatchSize      int32
-	Comment        bsoncore.Value
-	MaxTimeMS      int64
-	Limit          int32
-	CommandMonitor *event.CommandMonitor
-	Crypt          Crypt
-	ServerAPI      *ServerAPIOptions
+	BatchSize             int32
+	Comment               bsoncore.Value
+	MaxTimeMS             int64
+	Limit                 int32
+	CommandMonitor        *event.CommandMonitor
+	Crypt                 Crypt
+	ServerAPI             *ServerAPIOptions
+	MarshalValueEncoderFn func(io.Writer) (*bson.Encoder, error)
+	Registry              *bsoncodec.Registry
```

Comment:
Optional: You can remove the need to add `Registry` to `CursorOptions` and `BatchCursor` by incorporating it into the `MarshalValueEncoderFn`. See related suggestion on `MarshalValue` and `newEncoderFn`.